client/src/containers/Dataset.jsx

```diff
@@ -49,31 +51,24 @@ class Dataset extends Component {
       <div className="Dataset">
         <DatasetHeader
           onShowDatasetSettings={this.handleShowDatasetSettings}
-          name={dataset.name}
-          id={dataset.id}
+          name={getTitle(dataset)}
+          id={getId(dataset)}
         />
-        {dataset.rows != null ?
+        {getRows(dataset) != null &&
           <DatasetTable
-            columns={dataset.columns}
-            rows={dataset.rows}
-            transformations={dataset.transformations}
-            onTransform={(transformation) => dispatch(transform(dataset.id, transformation))}
-            onUndoTransformation={() => dispatch(undoTransformation(dataset.id))}
-          />
-          :
-          <div>loading...</div>}
+            columns={getColumns(dataset)}
+            rows={getRows(dataset)}
+            transformations={getTransformations(dataset)}
+            onTransform={(transformation) => dispatch(transform(getId(dataset), transformation))}
+            onUndoTransformation={() => dispatch(undoTransformation(getId(dataset)))}
+          />}
       </div>
     );
   }
 }
 
 Dataset.propTypes = {
-  dataset: PropTypes.shape({
-    id: PropTypes.string.isRequired,
-    name: PropTypes.string.isRequired,
-    columns: PropTypes.array,
-    rows: PropTypes.array,
-  }),
+  dataset: PropTypes.object.isRequired,
```

Comment:
This gives an error 'Required prop `dataset` was not specified in `Dataset`.' when cold-loading a saved dataset in the dataset-editor